### PR TITLE
DEV-949 argocd tool

### PR DESCRIPTION
**Pull request by EvgeniyReich**

### CODE REVIEW

**Comment by EvgeniyReich** [comment]
added the diff,set and sync tools and an updated tf configuration for an agent with all three tools.

File: argocd/argocd-agent-tf/main.tf [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  //Optional Fields (omitting will retain the current values): 
+  secrets      = ["ARGOCD_PASSWORD"]
+  integrations = var.agent_integrations
+  users        = ["evgeniy.reichelson@kubiya.ai"]
+  groups       = ["Admin"]
+  links        = []
+  tool_sources = var.agent_tool_sources
+
+  //Objects
+  # starters = [
```
Comment (koss110) [inline argocd/argocd-agent-tf/main.tf:23]: uncomment and leave empty lists or remove @EvgeniyReich

File: argocd/argocd-agent-tf/main.tf [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  #     name = "Check pod status"
+  #     command      = "kubectl get pods"
+  #   },
+  #   {
+  #     name = "Scale deployment"
+  #     command      = "kubectl scale deployment my-deployment --replicas=3"
+  #   }
+  # ]
+
+  # tasks = [
```
Comment (koss110) [inline argocd/argocd-agent-tf/main.tf:34]: same here @EvgeniyReich

File: argocd/argocd-tools/argo-diff-tool/argo-diff-tool.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        argocd app diff "$app_name"
+    else
+        argocd app diff $app_name --revision $revision
+    fi
+    return 0
+  args:
+    app_name:
+      description: 'The name of the live app to compare. Use default value if no other is specified'
+      required: false
+      default: "koss110-client"
```
Comment (koss110) [inline argocd/argocd-tools/argo-diff-tool/argo-diff-tool.yaml:23]: leave empty string or remove @EvgeniyReich

File: argocd/argocd-tools/argo-set-tool/argo-set-tool.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    # Login first
+    argocd login $ARGOCD_SERVER --username $ARGOCD_USERNAME --password $ARGOCD_PASSWORD  --insecure  --plaintext
+    # Set the app to a different revision
+     argocd app set "{{ .app_name }}" --revision "{{ .revision }}"
+    return 0
+  args:
+    app_name:
+      description: 'The name of the live app to set to a new revision. Use default value if no other is specified'
+      required: false
+      default: "koss110-client"
```
Comment (koss110) [inline argocd/argocd-tools/argo-set-tool/argo-set-tool.yaml:17]: same here

File: argocd/argocd-tools/argo-sync-tool/argo-sync-tool.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    done
+
+    if [ "$user_allowed" = false ]; then
+        echo "Sorry, you are not allowed to sync."
+    fi
+  args:
+    app_name:
+      description: 'The name of the live app to sync. Use default value if no other is specified'
+      required: false
+      default: "koss110-client"
```
Comment (koss110) [inline argocd/argocd-tools/argo-sync-tool/argo-sync-tool.yaml:31]: and here

File: argocd/argocd-agent-tf/main.tf [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+  //Optional fields, String
+  model = var.agent_llm_model // If not provided, Defaults to "azure/gpt-4"
+  //If not provided, Defaults to "ghcr.io/kubiyabot/kubiya-agent:stable"
+  image = var.agent_image
+
+  //Optional Fields (omitting will retain the current values): 
+  secrets      = ["ARGOCD_PASSWORD"]
+  integrations = var.agent_integrations
+  users        = ["evgeniy.reichelson@kubiya.ai"]
```
Comment (koss110) [inline argocd/argocd-agent-tf/main.tf:17]: leave empty @EvgeniyReich

**Review by koss110 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**